File: ShearNDipityLib/src/Models/Availability.java
package Models;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;


/**
 * Created with IntelliJ IDEA.
 * User: mindikingsun
 * Date: 29/08/13
 * Time: 1:05 PM
 * To change this template use File | Settings | File Templates.
 */
public class Availability {

    private int providerId;
    private String date;
    private String start;
    private String end;
    private String color;

    public Availability() {
    }

    public Date getDate() {
        try {
            return new SimpleDateFormat("yyyy-MM-dd").parse(this.date);
        } catch (ParseException e) {
            return null;
        }
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setDate(java.util.Date date) {
        this.date = new SimpleDateFormat("yyyy-MM-dd").format(date);
    }

    public int getProviderId() {
        return providerId;
    }

    public void setProviderId(int providerId) {
        this.providerId = providerId;
    }

    public String getStart() {
        return start;
    }

    public void setStart(String start) {
        this.start = start;
    }

    public String getEnd() {
        return end;
    }

    public void setEnd(String end) {
        this.end = end;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public Date getStartDate() throws ParseException {
        return new SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.ENGLISH).parse(date + " " + start);
    }

    public Date getEndDate() throws ParseException {
        return new SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.ENGLISH).parse(date + " " + end);
    }
}

File: Server/server/models/Availability.js
var Ring = require('ring');
var Utilities = require('../utilities/Utilities.js');

var Availability = Ring.create({
    _providerId: 0,
    _date: "",
    _start: "00:00",
    _end: "23:45",
    _color: "",

    init: function () {
    },

    setId: function (value) {
        if (Utilities.isIntegerAboveZero(value)) {
            this._providerId = value;
        }
    },
    getId: function () {
        return this._providerId;
    },
    setProviderId: function (value) {
        if (Utilities.isIntegerAboveZero(value)) {
            this._providerId = value;
        }
    },
    setColor: function (value) {
        if (Utilities.isStringAndNotEmpty(value)) {
            this._color = value;
        }
    },
    getColor: function () {
        return this._color;
    },
    getProviderId: function () {
        return this._providerId;
    },
    setDate: function (value) {
        this._date = value;
    },
    getDate: function () {
        return this._date;
    },
    setStart: function (value) {
        if (Utilities.isTimeStamp(value)) {
            this._start = value;
        }
    },
    getStart: function () {
        return this._start;
    },

    setEnd: function (value) {
        if (Utilities.isTimeStamp(value)) {
            this._end = value;
        }
    },
    getEnd: function () {
        return this._end;
    },
    toJSON: function () {
        return {
            "providerId": this._providerId,
            "date": this._date,
            "start": this._start,
            "end": this._end,
            "color": this._color
        }
    }
});

module.exports = Availability;